boards/nucleo-f334r8/include/periph_conf.h

```diff
@@ -79,8 +101,13 @@ static const uart_conf_t uart_config[] = {
         .rx_af      = GPIO_AF7,
         .tx_af      = GPIO_AF7,
         .bus        = APB2,
-        .irqn       = USART1_IRQn
+        .irqn       = USART1_IRQn,
+#ifdef MODULE_PERIPH_DMA
+        .dma        = 2,
+        .dma_chan   = DMA_CHAN_CONFIG_UNSUPPORTED
+#endif
     },
+#if !defined(MODULE_PERIPH_SPI) || !defined(MODULE_PERIPH_DMA)
```

Comment:
Maybe add a comment that describes why the third UART is disabled in this case. Something like: "SPI1 RX and USART3 are sharing the same DMA channel, so disable the third UART when both SPI and DMA features are enabled."